pkg/controller/cloudfront/distribution/lateinit.go

```diff
@@ -728,3 +737,50 @@ func lateInitLambdaFunctionAssociation(in *svcapitypes.LambdaFunctionAssociation
 	in.IncludeBody = awsclients.LateInitializeBoolPtr(in.IncludeBody, from.IncludeBody)
 	in.LambdaFunctionARN = awsclients.LateInitializeStringPtr(in.LambdaFunctionARN, from.LambdaFunctionARN)
 }
+
+func lateInitFunctionAssociations(in *svcapitypes.FunctionAssociations, from *svcsdk.FunctionAssociations) {
+	if len(from.Items) == 0 {
+		return
+	}
+
+	// If we have no function associations, late init the entire slice
+	if in.Items == nil {
+		in.Items = make([]*svcapitypes.FunctionAssociation, len(from.Items))
+		for i := range from.Items {
+			in.Items[i] = &svcapitypes.FunctionAssociation{}
+			lateInitFunctionAssociation(in.Items[i], from.Items[i])
+		}
+
+		return
+	}
+
+	// If we have some function associations, we need to late init each one of them
+	existing := make(map[string]*svcsdk.FunctionAssociation)
+	for i := range from.Items {
+		o := from.Items[i]
+		if o.FunctionARN == nil {
+			continue
+		}
+		// FunctionARN must be unique for each FunctionAssociation
+		existing[awsclients.StringValue(o.FunctionARN)] = o
+	}
+
+	for i := range in.Items {
+		il := in.Items[i]
+		if il.FunctionARN == nil {
```

Comment:
If `il.FunctionARN` is `nil`, should we skip it or actually late-initialize it (set it to `fl.FunctionARN`)? Doesn't a `nil` value denote an unset value in this context?